### PR TITLE
The structure of the Extension

### PR DESCRIPTION
**Pull request by 928PJY**
The structure of extension

### CODE REVIEW

File: src/VscPreviewExtension/.gitignore [hunk 1]
```diff
@@ -0,0 +1,6 @@
+node_modules/
+out/
+src/*
```
Comment (qinezh) [inline src/VscPreviewExtension/.gitignore]: why need `*` here but no need it in above case?

Comment (928PJY) [inline src/VscPreviewExtension/.gitignore]: I wanted to add the folder but not the files , but it didn't work

---
In reply to: [74028182](https://github.com/dotnet/docfx/pull/589#discussion_r74028182) [](ancestors = 74028182)

File: src/VscPreviewExtension/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
+{
+    "compilerOptions": {
```
Comment (hellosnow) [inline src/VscPreviewExtension/tsconfig.json]: >compilerOptions [](start = 5, length = 15)

indent of *.json file better be 2 spaces? can config in VS

File: src/VscPreviewExtension/package.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
+{
+    "name": "DfmPreview",
+    "displayName": "DfmPreview",
+    "description": "preview the DocFX Flavored Markdown in vscode",
```
Comment (hellosnow) [inline src/VscPreviewExtension/package.json]: >preview [](start = 20, length = 7)

Preview?

File: src/VscPreviewExtension/.gitignore [hunk 4]
```diff
@@ -0,0 +1,6 @@
...
+test/
+media/*
+DfmParse/*
```
Comment (hellosnow) [inline src/VscPreviewExtension/.gitignore]: how about we add the content later? seems none of the folder is added in this review?

File: src/VscPreviewExtension/package.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    ],
+    "main": "./out/src/extension",
+    "contributes": {
+        "commands": [
+            {
+                "command": "DFM.showpreview",
+                "title": "Toggle Preview"
+            },
+            {
+                "command": "DFM.showpreviewToside",
```
Comment (hellosnow) [inline src/VscPreviewExtension/package.json]: >showpreviewToside [](start = 32, length = 17)

naming better to camel "showPreviewToside" or all lowercase?

File: src/VscPreviewExtension/README.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    - `ctrl+shift+q`    Preview
+    - `ctrl+k q`        Preview to Side
+ 
+## How to install from Marketplace
+This extension is hosted at [Visual Studio Marketplace](http://marketplace.visualstudio.com/items?itemName=928pjy.DfmPreview)
+> **Notice** Upgrade to Visual Studio Code 1.3.0 or above.
+
+## Change Log
+### 0.0.9
+* Bug fix: Can't preview a single file(not in a open folder)
```
Comment (superyyrrzz) [inline src/VscPreviewExtension/README.md]: >C [](start = 11, length = 1)

lower case after colon?

File: src/VscPreviewExtension/README.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+* Bug fix: throw an error if it is not a markdown file
+
+### 0.0.8
+* Bug fix: Open other preview window
```
Comment (superyyrrzz) [inline src/VscPreviewExtension/README.md]: >O [](start = 11, length = 1)

lower case after colon?

**Pull request closed without merging**